packages/botonic-plugin-flow-builder/src/webview/use-webview-contents.ts

```diff
@@ -31,6 +31,28 @@ export function useWebviewContents<T extends MapContentsType>({
   const [isLoading, setLoading] = useState(true)
   const [error, setError] = useState(false)
 
+  const updateCurrentLocale = (textContents?: WebviewTextContent[]) => {
+    if (
+      textContents &&
+      textContents?.length > 0 &&
+      textContents[0].content.text.some(text => text.locale === currentLocale)
+    ) {
+      setCurrentLocale(currentLocale)
+      return
+    }
+
+    const language = currentLocale.split('-')[0]
+    if (textContents?.[0].content.text.some(text => text.locale === language)) {
+      setCurrentLocale(language)
+      return
+    }
+
+    console.error(
+      `locale: ${currentLocale} cannot be resolved with: ${textContents?.[0].content.text.map(text => text.locale).join(', ')}`
+    )
+    return
+  }
```

Comment:
Do happy path and early returns
```suggestion
const updateCurrentLocale = (textContents?: WebviewTextContent[]) => {
  if (!textContents || textContents.length === 0) {
    return;
  }

  const textContents = textContents[0].content.text;
  
  if (textContents.some(text => text.locale === currentLocale)) {
    setCurrentLocale(currentLocale);
    return;
  }
  
  const language = currentLocale.split('-')[0];
  if (textContents.some(text => text.locale === language)) {
    setCurrentLocale(language);
    return;
  }

  console.error(
    `locale: ${currentLocale} cannot be resolved with: ${textArray.map(text => text.locale).join(', ')}`
  );
};
```

Reply:
applied